bo/reklamation.go

```diff
@@ -0,0 +1,21 @@
+package bo
+
+import (
+	"github.com/hochfrequenz/go-bo4e/com"
+	"github.com/hochfrequenz/go-bo4e/enum/lokationstyp"
+	"github.com/hochfrequenz/go-bo4e/enum/reklamationsgrund"
+)
+
+// Reklamation wird benutzt um eine fehlende, oder unplausible Energiemenge anzuzeigen. Für Reklamationen von Rechnungen, wird ein Avis vom Typ Nicht-Zahlungsavis benutzt
+type Reklamation struct {
+	Geschaeftsobjekt
+	LokationsId  string                              `json:"lokationsId,omitempty" example:"DE0123456789012345678901234567890" validate:"alphanum,required,min=11,max=33"` // LokationsId is the ID of the location (either a LokationsTyp MALO ID (11 digits) or a LokationsTyp MELO ID (33 alphanum))
+	LokationsTyp lokationstyp.Lokationstyp           `json:"lokationsTyp,omitempty" example:"MELO" validate:"required"`                                                    // LokationsTyp is the type of the location in LokationsId
+	Grund        reklamationsgrund.Reklamationsgrund `json:"grund,omitempty" validate:"required"`                                                                          // The reason for the complaint
+	ObisKennzahl *string                             `json:"obisKennzahl,omitempty" validate:"required,omitempty" example:"1-0:1.8.1"`                                     // Die ObisKennzahl für das Zählwerk, die festlegt, welche auf die gemessene Größe mit dem Stand gemeldet wird. Nur Zählwerkstände mit dieser OBIS-Kennzahl werden an diesem Zählwerk registriert.
+	Zeitraum     *com.Zeitraum                       `json:"zeitraum,omitempty" validate:"required,omitempty"`                                                             // Zeitraum defines the com.Zeitraum where the Energiemenge is complained
```

Comment:
macht die reihenfolge der vaidators hier sinn?
Das hier hieße ja: Es ist required aber wenn es empty ist, dann überspring die validierung.
Bei der Obis könnte man sowas machen wie: "omitempty,validateobisregex" (wnen es das gäbe) aber required mit omitempty zu kombinieren ist meiner einschätzung nie sinnvoll, egal in welcher reihenfolge.

Reply:
So?
722f846c3abf5e2c111e63828fd8615b040dad97